srml/staking/src/lib.rs

```diff
@@ -1399,22 +1411,42 @@ impl<T: Trait> Module<T> {
 		}
 	}
 
-	/// Add reward points to validator.
+	/// Add reward points to validators using their stash account ID.
+	///
+	/// Validators are keyed by stash account ID and must be in the current elected set.
+	///
+	/// For each element in the iterator the given number of points in u32 is added to the
+	/// validator, thus duplicates are handled.
 	///
 	/// At the end of the era each the total payout will be distributed among validator
 	/// relatively to their points.
-	pub fn add_reward_points_to_validator(validator: T::AccountId, points: u32) {
-		<Module<T>>::current_elected().iter()
-			.position(|elected| *elected == validator)
-			.map(|index| {
-				CurrentEraRewards::mutate(|rewards| {
-					if let Some(new_total) = rewards.total.checked_add(points) {
-						rewards.total = new_total;
-						rewards.rewards.resize((index + 1).max(rewards.rewards.len()), 0);
-						rewards.rewards[index] += points; // Addition is less than total
+	pub fn add_reward_points(validators_points: Vec<(T::AccountId, u32)>) {
```

Comment:
why does it have to be a `Vec` and force allocation (expensive in runtime)? it should be a slice or better yet, an `IntoIterator`.

Reply:
rename to `reward_by_ids`?

Reply:
renamed and put intoiterator.

now we iterate on validators to reward first and inside this loop we iterate on current_elected.

Reply:
Also if this method is used for rewarding a lots of ppl at the same time we should do a sort of current_elected and then a binary_search.

hmm I should state that in the doc